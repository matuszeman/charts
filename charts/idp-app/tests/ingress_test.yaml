suite: ingress
templates:
  - ingress.yaml
values:
  - values/minimum.yaml
tests:
  - it: should match snapshot with values-ci.yaml
    values:
      - ../values-ci.yaml
    asserts:
      - matchSnapshot: {}
  - it: when ingress is enabled but not service
    values:
      - values/ingress-minimum.yaml
    asserts:
      - failedTemplate:
          errorMessage: "ingress: service must be enabled"
  - it: when multi deployment
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/multi.yaml
    asserts:
      - matchSnapshot: {}
  - it: when ingress and ingresses are set
    values:
      - values/deployment.yaml
      - values/service.yaml
    set:
      ingress:
        enabled: true
        className: nginx-external
        domain: example.com
        host: testapp-default
      ingresses:
        external:
          enabled: true
          className: nginx-external
          domain: example.com
          host: testapp-external
          oauth2ProxyAuth:
            enabled: true
            url: https://auth.example.com
            allowedGroups:
              - one@example.com
        internal:
          enabled: true
          domain: example.com
          host: testapp-internal
          accessLog:
            enabled: false
          basicAuth:
            enabled: true
            secretName: basic-auth


    asserts:
      - matchSnapshot: { }
  - it: can disable accessLog
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/ingress-minimum.yaml
    set:
      ingress:
        accessLog:
          enabled: false
    asserts:
      - isSubset:
          path: metadata.annotations
          content:
            nginx.ingress.kubernetes.io/enable-access-log: "false"
  - it: when oauth2ProxyAuth is enabled on path
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/ingress-minimum.yaml
    set:
      ingress:
        oauth2ProxyAuth:
          enabled: true
          service:
            name: my-oauth2-proxy
    asserts:
      - matchSnapshot: {}
  - it: when oauth2ProxyAuth is enabled on url
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/ingress-minimum.yaml
    set:
      ingress:
        oauth2ProxyAuth:
          enabled: true
          url: https://auth.example.com
          allowedGroups:
            - one@example.com
            - two@example.com
    asserts:
      - matchSnapshot: { }
  - it: when oauth2ProxyAuth with externalUrl
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/ingress-minimum.yaml
    set:
      ingress:
        oauth2ProxyAuth:
          enabled: true
          url: http://oauth2proxy.devsup.svc.cluster.local
          urlExternal: https://auth.example.com
    asserts:
      - matchSnapshot: { }
  - it: when oauth2ProxyAuth with templated externalUrl, then $ global var is resolvable
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/ingress-minimum.yaml
    set:
      ingress:
        oauth2ProxyAuth:
          enabled: true
          url: http://oauth2proxy.devsup.svc.cluster.local
          urlExternal: https://auth.{{ $.Values.global.cluster }}.example.com
    asserts:
      - isSubset:
          path: metadata.annotations
          content:
            nginx.ingress.kubernetes.io/auth-signin: "https://auth.test.example.com/oauth2/start?rd=https://$host$request_uri"
  - it: when oauth2ProxyAuth with templated externalUrl, then .cluster var is resolvable
    values:
      - values/deployment.yaml
      - values/service.yaml
      - values/ingress-minimum.yaml
    set:
      global:
        idpAppConfig:
          clusters:
            test:
              oauth2ProxyAuth:
                urlExternal: https://auth.test-222.example.com
      ingress:
        oauth2ProxyAuth:
          enabled: true
          url: http://oauth2proxy.devsup.svc.cluster.local
          urlExternal: "{{ .cluster.oauth2ProxyAuth.urlExternal }}"
    asserts:
      - isSubset:
          path: metadata.annotations
          content:
            nginx.ingress.kubernetes.io/auth-signin: "https://auth.test-222.example.com/oauth2/start?rd=https://$host$request_uri"
  - it: when both service.enabled and ingress.service is set, then fail
    values:
      - values/deployment.yaml
      - values/service.yaml
    set:
      ingress:
        enabled: true
        className: nginx-internal
        domain: example.com
        host: existing-service
        service:
          name: existing-service
          port: 80
    asserts:
      - failedTemplate:
          errorMessage: "ingress: both service.enabled and ingress.service is set"
  - it: when service is disabled and ingress service is set
    set:
      ingress:
        enabled: true
        className: nginx-internal
        domain: example.com
        host: existing-service
        service:
          name: existing-service
          port: 80
    asserts:
      - matchSnapshot: { }
  - it: ingress service is templated
    set:
      ingress:
        enabled: true
        className: nginx-internal
        domain: example.com
        host: existing-service
        service:
          name: "{{ .Release.Name }}-existing-service"
          port: 80
    asserts:
      - isSubset:
          path: spec.rules[0].http.paths[0].backend.service
          content:
            name: RELEASE-NAME-existing-service
            port:
              number: 80