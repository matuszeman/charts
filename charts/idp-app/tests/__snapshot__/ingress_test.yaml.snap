should match snapshot with values-ci.yaml:
  1: |
    apiVersion: v1
    data:
      SCRIPT_FILENAME: /html$fastcgi_script_name
    kind: ConfigMap
    metadata:
      name: RELEASE-NAME-resource-suffix-ingress-fcgi
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/auth-realm: Authentication Required
        nginx.ingress.kubernetes.io/auth-secret: test-basic-auth
        nginx.ingress.kubernetes.io/auth-type: basic
        nginx.ingress.kubernetes.io/backend-protocol: FCGI
        nginx.ingress.kubernetes.io/enable-access-log: "true"
        nginx.ingress.kubernetes.io/fastcgi-index: index.php
        nginx.ingress.kubernetes.io/fastcgi-params-configmap: RELEASE-NAME-resource-suffix-ingress-fcgi
        nginx.ingress.kubernetes.io/rewrite-target: /api/$2
        nginx.ingress.kubernetes.io/use-regex: true
      labels:
        app.kubernetes.io/instance: RELEASE-NAME-resource-suffix
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-resource-suffix
    spec:
      ingressClassName: nginx-external
      rules:
        - host: test.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-resource-suffix
                    port:
                      name: app
                path: /somepath(/|$)(.*)
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - test.example.com
          secretName: RELEASE-NAME-resource-suffix-ingress-tls
when ingress and ingresses are set:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      ingressClassName: nginx-external
      rules:
        - host: testapp-default.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - testapp-default.example.com
          secretName: RELEASE-NAME-ingress-tls
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/auth-response-headers: Auth-Proxy-Email,Auth-Proxy-Groups
        nginx.ingress.kubernetes.io/auth-signin: https://auth.example.com/oauth2/start?rd=https://$host$request_uri
        nginx.ingress.kubernetes.io/auth-snippet: |
          proxy_set_header X-Forwarded-Host testapp-external.example.com;
        nginx.ingress.kubernetes.io/auth-url: https://auth.example.com/oauth2/auth?allowed_groups=one@example.com
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-external
    spec:
      ingressClassName: nginx-external
      rules:
        - host: testapp-external.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - testapp-external.example.com
          secretName: RELEASE-NAME-external-ingress-tls
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/auth-realm: Authentication Required
        nginx.ingress.kubernetes.io/auth-secret: basic-auth
        nginx.ingress.kubernetes.io/auth-type: basic
        nginx.ingress.kubernetes.io/enable-access-log: "false"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-internal
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: testapp-internal.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - testapp-internal.example.com
          secretName: RELEASE-NAME-internal-ingress-tls
when multi deployment:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-one
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: one.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-one
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - one.example.com
          secretName: RELEASE-NAME-one-ingress-tls
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-two
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: two.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME-two
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - two.example.com
          secretName: RELEASE-NAME-two-ingress-tls
when oauth2ProxyAuth is enabled on path:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/auth-response-headers: Auth-Proxy-Email,Auth-Proxy-Groups
        nginx.ingress.kubernetes.io/auth-signin: https://$host/oauth2/start?rd=https://$host$request_uri
        nginx.ingress.kubernetes.io/auth-snippet: |
          proxy_set_header X-Forwarded-Host testapp.example.com;
        nginx.ingress.kubernetes.io/auth-url: https://$host/oauth2/auth
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: testapp.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - testapp.example.com
          secretName: RELEASE-NAME-ingress-tls
  2: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-oauth2-proxy
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: testapp.example.com
          http:
            paths:
              - backend:
                  service:
                    name: my-oauth2-proxy
                    port:
                      number: 80
                path: /oauth2
                pathType: Prefix
      tls:
        - hosts:
            - testapp.example.com
          secretName: RELEASE-NAME-ingress-tls
when oauth2ProxyAuth is enabled on url:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/auth-response-headers: Auth-Proxy-Email,Auth-Proxy-Groups
        nginx.ingress.kubernetes.io/auth-signin: https://auth.example.com/oauth2/start?rd=https://$host$request_uri
        nginx.ingress.kubernetes.io/auth-snippet: |
          proxy_set_header X-Forwarded-Host testapp.example.com;
        nginx.ingress.kubernetes.io/auth-url: https://auth.example.com/oauth2/auth?allowed_groups=one@example.com,two@example.com
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: testapp.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - testapp.example.com
          secretName: RELEASE-NAME-ingress-tls
when oauth2ProxyAuth with externalUrl:
  1: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      annotations:
        cert-manager.io/cluster-issuer: default
        nginx.ingress.kubernetes.io/auth-response-headers: Auth-Proxy-Email,Auth-Proxy-Groups
        nginx.ingress.kubernetes.io/auth-signin: https://auth.example.com/oauth2/start?rd=https://$host$request_uri
        nginx.ingress.kubernetes.io/auth-snippet: |
          proxy_set_header X-Forwarded-Host testapp.example.com;
        nginx.ingress.kubernetes.io/auth-url: http://oauth2proxy.devsup.svc.cluster.local/oauth2/auth
        nginx.ingress.kubernetes.io/enable-access-log: "true"
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      ingressClassName: nginx-internal
      rules:
        - host: testapp.example.com
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      name: app
                path: /
                pathType: ImplementationSpecific
      tls:
        - hosts:
            - testapp.example.com
          secretName: RELEASE-NAME-ingress-tls
